### ISSUE TITLE
SWITCH cannot be used Statically - incorrect

### ISSUE THREAD

**Issue by edhans** [tabulartools/dax-optimizer#2]
### Describe the problem

Without posting the entire measure, this is the critical part. The entire SWITCH variable was highlighted as red with this error.

```
VAR varSelectedSource = [Comparison Key]
...omitted code...
VAR Result =
    // see comments in varWaterfallCompareRowCount for the alternate value at end of SWITCH
    SWITCH(
        varSelectedSource,
        "Snapshot 1", varSnapshot1ValueAssigned,
        "Snapshot 2", varSnapshot2ValueAssigned,
        "Diff", varDifference,
        "Diff%", varDifferencePct,
        DIVIDE(
            ABS( varDifference ),
            varWaterFallCompareRowCount > 0
        )
    )
RETURN
```

The value returned from the [Comparison Key] measure (which is just a SELECTEDVALUE(Table[FieldIntheVisual]) ) is the field being filtered in the model, so this should not have thrown that error.
![image](https://github.com/tabulartools/dax-optimizer/assets/45298358/c62b0454-0f50-4ee0-9805-221fcfeeea74)

**Comment by marcosqlbi** [comment]
The SWITCH issue is correct based on our assumptions: if the column Comparison Type is hidden, we assume that it cannot be used by the end user and another column will be filtered so that the SWITCH optimization at the query plan level is not applied. 
For this reason, we wrote in https://kb.daxoptimizer.com/d/102200 :

_To optimize the query plan, [SWITCH](https://dax.guide/switch/) should test the column that is directly filtered. For example, if a visible column is ordered by a hidden column, the [SWITCH](https://dax.guide/switch/) function should use the visible column and not the hidden column._

There are additional considerations about the variables' scope that are not implemented yet, but that are in our backlog. This measure should report a potential optimization for the variables, which would end up requiring a nested SWITCH that executes Snapshot 1 / 2 only when the selection is not "Diff" or "Diff%". We took note of this example to make sure we cover it correctly once we implement the new rule.

Thanks!
